models/classes/TestCategoryPresetProvider.php

```diff
@@ -18,78 +18,88 @@
  * Copyright (c) 2017 (original work) Open Assessment Technologies SA;
  */
 
+declare(strict_types=1);
+
 /**
  * @author Christophe Noël <christophe@taotesting.com>
  */
 
 namespace oat\taoQtiTest\models;
 
-
 use oat\oatbox\service\ConfigurableService;
 use oat\taoTests\models\runner\plugins\TestPluginService;
+use RuntimeException;
 
 class TestCategoryPresetProvider extends ConfigurableService
 {
-    const SERVICE_ID = 'taoQtiTest/CategoryPresetProvider';
+    public const SERVICE_ID = 'taoQtiTest/CategoryPresetProvider';
 
-    const GROUP_NAVIGATION  = 'navigation';
-    const GROUP_WARNING     = 'warning';
-    const GROUP_TOOLS       = 'tools';
+    public const GROUP_NAVIGATION = 'navigation';
+    public const GROUP_WARNING    = 'warning';
+    public const GROUP_TOOLS      = 'tools';
 
     private $allPresets;
 
+    private $isGroomed = false;
+
     /**
      * TestCategoryPresetProvider constructor.
+     *
      * @param array $options
      * @param array $allPresets - allow override of preset list
      */
-    public function __construct(array $options = [], $allPresets = [])
+    public function __construct(array $options = [], array $allPresets = [])
     {
         $this->allPresets = $allPresets;
 
         parent::__construct($options);
     }
 
-    protected function getPresetGroups()
+    /**
+     * @param string               $presetGroup
+     * @param TestCategoryPreset[] $presets
+     */
+    public function register(string $presetGroup, array $presets): void
```

Comment:
This method signature change is a breaking change for our customer extension. I will create PR with fix. But would be good to check all other usages.

Reply:
I wasn't expecting anyone to use this "feature", accepting a single preset :/
Reverting, thanks!